### ISSUE TITLE
Some question about code

### ISSUE THREAD

**Issue by LC-HX** [cocagne/multi-paxos-example#12]
there are lots of methods which call the methods of superclass that there exactly isn't this method in the superclass and the names of these two methods are always the same. Is this a special feature for Python? 
eg. super(SimpleSynchronizationStrategyMixin,self).set_messenger(messenger) 
the superclass of SimpleSynchronizationStrategyMixin is object not containing the method "set_messenger". Exactly the name of method containing this line of code is also "set_messenger"

**Comment by cocagne** [comment]
Yeah, read the section in the docs about mix in classes, it's an unusual
pattern in most languages and relies on dynamic method lookup at runtime.
Very confusing at first but it affords a great deal of flexibility and code
reuse.
